supervisor/api/utils.py

```diff
@@ -116,6 +118,28 @@ async def wrap_api(api, *args, **kwargs):
     return wrap_method
 
 
+def api_process_custom(content_type):
```

Comment:
This annotation looks kind of redundant `api_process_raw` to be honest. It's set up exactly the same way except this one only actually supports content type text since it always returns errors as plain text, regardless of the specified content type. Whereas `api_process_raw` only actually supports content type binary because it always encodes errors as binary, regardless of the specified content type.

What if we just did this instead?
```py
def api_process_raw(content, *, error_type=None):
    """Wrap content_type into function."""

    def wrap_method(method):
        """Wrap function with raw output to rest api."""
        async def wrap_api(api, *args, **kwargs):
            """Return api information."""
            try:
                msg_data = await method(api, *args, **kwargs)
            except HassioError as err:
                return api_return_error(err, error_type=error_type or CONTENT_TYPE_BINARY)

            if isinstance(msg_data, (web.Response, web.StreamResponse)):
                return msg_data
            return web.Response(body=msg_data, content_type=content)

def api_return_error(
    error: Exception | None = None, message: str | None = None, error_type: str | None = None
) -> web.Response:
    """Return an API error message."""
    if error and not message:
        message = get_message_from_exception_chain(error)
        if check_exception_chain(error, DockerAPIError):
            message = format_message(message)
    if not message:
        message = "Unknown error, see supervisor"
    
    status = 400
    if (is_api_error := isinstance(error, APIError)):
        status = error.status

    match error_type:
        case CONTENT_TYPE_TEXT:
            return web.Response(body=message, content_type=error_type, status=status)
        case CONTENT_TYPE_BINARY:
            return web.Response(body=message.encode(), content_type=error_type, status=status)
        case _:
            result = {
                JSON_RESULT: RESULT_ERROR,
                JSON_MESSAGE: message,
            }
            if is_api_error and error.job_id:
                result[JSON_JOB_ID] = error.job_id

    return web.json_response(
        result,
        status=status,
        dumps=json_dumps,
    )
```
This has a couple benefits:
1. `api_process_raw` works and we don't need to create a new `api_process_custom`
2. `api_process_raw` actually supports content type text and binary and isn't just secretly binary only because of how it does error
3. Shared error handling in all our annotations. Including how we make the message from the exception (since that is currently quite different in `api_process_raw` and `api_process_custom` which doesn't seem right)

Reply:
This is definitely better - and something I also wanted to do, I was just hesitant to make larger adjustments to it - for example I didn't know why it had `msg_data` set to empty string for `HassioError` and returned message text for `APIError, APIForbidden`. So I agree, let's go with your suggestion, thanks!

(I just needed to do adjust the match cases to dotted attributes because the suggested syntax is invalid, and reformatted the code.)